### PR TITLE
Issue/create a team/3

### LINKED ISSUE #42: Create Team
What has to be done:
Create a team and store the team in database

Why it has to be done:
So that the team captain can invite team players and join tournaments

When the task is done:
The frontend application can call the backend application with all information to create a team.
When the team name was already used it must return an error.
When the team name was new, it creates a record in the database.

### PR DESCRIPTION
**Pull request by lmtjanssen**
closes #42

### CODE REVIEW

**Review by lfb0801 — CHANGES_REQUESTED** [review]
Currently this pull request doesn't have any test.
Suggestion would be to test the service and leave the queries/mutations as they are basically endpoints for now.

**Comment by MindHawk** [comment]
We are using an in memory (H2) database for now, do we want to commit this to master or establish the Azure connection before committing? Else a task should be made for this.

File: src/main/java/com/fontys/api/mutations/TeamMutation.java [hunk 1]
```diff
@@ -3,16 +3,19 @@
...
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class TeamMutation implements GraphQLMutationResolver {
 
-    @Autowired
-    private TeamService teamService;
+    private final
+    TeamService teamService;
```
Comment (MindHawk) [inline src/main/java/com/fontys/api/mutations/TeamMutation.java:12]: Why is this not on the same line

**Review by larsvandriel — COMMENTED** [review]
I should advise adding a field to the team class to store the team captain. That way we can always find out who the team captain is.

**Review by MindHawk — APPROVED** [review]
Good enough for me

**Review by lfb0801 — APPROVED** [review]
Looks good,
the fix for tests by making a mockRepository isn't the best case scenario, but for now it works

**Pull request merged**